charts/k8s-monitoring/templates/agent_config/_receivers.river.txt

```diff
@@ -0,0 +1,70 @@
+{{ define "agent.config.receivers" }}
+// Receivers
+{{- if (or .Values.receivers.grpc.enabled .Values.receivers.http.enabled) }}
+otelcol.receiver.otlp "receiver" {
+  debug_metrics {
+    {{- if or .Values.receivers.grpc.disable_debug_metrics .Values.receivers.http.disable_debug_metrics }}
+    disable_high_cardinality_metrics = true
+    {{- else }}
+    disable_high_cardinality_metrics = false
+    {{- end }}
+  }
+
+{{- if or .Values.receivers.port .Values.receivers.grpc.enabled }}
+{{$data := dict "port" (.Values.receivers.port | default .Values.receivers.grpc.port) "type" "OTLP gRPC" "agent" (index .Values "grafana-agent").agent }}
+{{- include "checkforAgentPort" $data }}
+  grpc {
+    endpoint = "0.0.0.0:{{ $data.port | int }}"
+  }
+{{- end }}
+{{- if .Values.receivers.http.enabled }}
+{{$data := dict "port" .Values.receivers.http.port "type" "OTLP HTTP" "agent" (index .Values "grafana-agent").agent }}
+{{- include "checkforAgentPort" $data }}
+  http {
+    endpoint = "0.0.0.0:{{ $data.port | int }}"
+  }
+{{- end }}
+  output {
```

Comment:
We have identified with @mar4uk and @rlankfo that a few transformation are needed when converting OTLP signals to LGTM:

Add K8s contextual attributes, including as Liki labels for compatibility with Grafana K8s monitoring:

```
 otelcol.processor.k8sattributes "default" {
    extract {
      metadata = [
        "k8s.namespace.name",
        "k8s.pod.name",
        "k8s.deployment.name",
        "k8s.statefulset.name",
        "k8s.daemonset.name",
        "k8s.cronjob.name",
        "k8s.job.name",
        "k8s.node.name",
        "k8s.pod.uid",
        "k8s.pod.start_time",
      ]
    }
    pod_association {
      source {
        from = "connection"
      }
    }
    
    output {
      logs    = [otelcol.processor.transform.add_k8s_attributes.input]
      traces  = [otelcol.processor.transform.add_k8s_attributes.input]
    }
  }
  otelcol.processor.transform "add_k8s_attributes" {
    // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.transform/
    // Grafana Cloud Kubernetes monitoring expects Loki labels `cluster`, `pod`, and `namespace`
    error_mode = "ignore"
    log_statements {
      context = "resource"
      statements = [
        "set(attributes[\"pod\"], attributes[\"k8s.pod.name\"])",
        "set(attributes[\"namespace\"], attributes[\"k8s.namespace.name\"])",
        "set(attributes[\"loki.resource.labels\"], \"pod, namespace, cluster, job\")",
      ]
    }
    output {
      logs    = [otelcol.processor.batch.default.input]
      traces  = [otelcol.processor.batch.default.input]
    }
  }
```


Add a few labels to Prometheus metrics to enable slicing & dicing metrics:

```
  otelcol.processor.transform "add_resource_attributes_as_metric_attributes" {
    // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.transform/
    error_mode = "ignore"
    metric_statements {
      context = "datapoint"
      statements = [
        "set(attributes[\"deployment.environment\"], resource.attributes[\"deployment.environment\"])",
        "set(attributes[\"service.version\"], resource.attributes[\"service.version\"])",
      ]
    }
    output {
      metrics = [otelcol.processor.batch.default.input]
    }
  }
```

Reply:
I'm so glad you added these! I definitely want to include them in this PR, since it'll be important to actually make this work successful.

Let me include these changes and I'll ping you for a re-review.